src/pages/quiz/QuizVariants.tsx

```diff
@@ -122,14 +120,15 @@ const ActiveQuiz: React.FC<ActiveQuizProps> = ({ finishQuiz }) => {
     setIsOpen(false)
   }, [])
 
-  const handleFinish = useCallback(async () => {
+  const handleFinish = async () => {
     const finishedQuiz = await mutateAsync()
-    setIsOpen(false)
-    finishQuiz(finishedQuiz?._id)
-    if (!scoredResponses) {
-      navigate(-1)
-    }
-  }, [finishQuiz, mutateAsync, navigate, scoredResponses])
+    navigate(
+      getFullUrl({
+        pathname: authRoutes.cooperationQuizReview.route,
+        parameters: { id: cooperationId, quizId, attemptId: finishedQuiz._id }
+      })
+    )
+  }
```

Comment:
Please take into account `scoredResponses` setting. I'm not sure that we should show responses in every case. Probably it should be something like this:
```suggestion
  if (scoredResponses) {
      navigate(
        getFullUrl({
          pathname: authRoutes.cooperationQuizReview.route,
          parameters: { id: cooperationId, quizId, attemptId: finishedQuiz._id }
        })
      )
    } else {
      navigate(-1)
    }
  }
```